xml/src/main/java/org/ehcache/xml/ConfigurationParser.java

```diff
@@ -132,6 +156,35 @@ public ConfigurationParser(String xml) throws IOException, SAXException, JAXBExc
     DocumentBuilder domBuilder = factory.newDocumentBuilder();
     domBuilder.setErrorHandler(new FatalErrorHandler());
     Element dom = domBuilder.parse(xml).getDocumentElement();
+
+    Stack<NodeList> nodeLists = new Stack<NodeList>();
+    nodeLists.push(dom.getChildNodes());
+    while (!nodeLists.isEmpty()) {
```

Comment:
I thought that method was a mess already, bunch of other are good candidates for becoming dedicated methods... But true, shouldn't be adding to the mess. Who wrote that crap?! Oh. Y, I think I started that ;)